File: Basic/Examples.fs
module Examples

let main() =
    let x = 5

    let list = [1; 2; 3; 4; 5]

    let doubled =
        list
        |> List.filter (fun x -> x > 1)
        |> List.map (fun x -> x * 2)

    let name: string option = None

    if name.IsSome then
        printfn "%d" name.Value.Length
    else
        ()

    let point = 3, 4
    let xCoord, yCoord = point
    printfn "Point: %d, %d" xCoord yCoord

    let swapped = yCoord, xCoord
    printfn "Swapped: %d, %d" (fst swapped) (snd swapped)

    let pair = "hello", 42
    let message, number = pair
    printfn "Pair: %s = %d" message number

let add a b = a + b
let add (a, b) = a + b

type User = { Name: string }

File: Basic/Examples.ts
class Examples {
  static main() {
    let x = 5;

    let list = [1, 2, 3, 4, 5];

    const doubled =
      list
      .filter((x) => x > 1)
      .map((x) => x * 2);

    let name: string | null = null;

    if (name !== null) {
      console.log(name.length);
    }

    const point: [number, number] = [3, 4];
    const [xCoord, yCoord] = point;
    console.log(`Point: ${xCoord}, ${yCoord}`);

    const swapped: [number, number] = [yCoord, xCoord];
    console.log(`Swapped: ${swapped[0]}, ${swapped[1]}`);

    const pair: [string, number] = ["hello", 42];
    const [message, number] = pair;
    console.log(`Pair: ${message} = ${number}`);
  }

  add(a: number, b: number): number {
    return a + b;
  }
}

interface User {
  Name: string;
}
